### ISSUE TITLE
First and second stage refactor

### ISSUE THREAD

**Issue by castagninojose** [castagninojose/imgalaxy#14]
Refactor the training codebase such that the first and second stages are more transparent to use, allowing testing models with the least amount of changes.

**Comment by castagninojose** [comment]
With changes introduced in #12 the model now runs for multiple masks but for some reason launching a sweep (second stage) produces a `tf.ops` error
```
cannot compute ConcatV2 as input #1(zero-based) was expected to be a float tensor but is a half tensor
```
specifically in the augmentation layer, when we concat images and masks. This error is typical when trying to concat two tensors with a dtype mismatch. In our case, launching a sweep causes tensorflow to use an integer version of the mask, even when it is explicitly cast as a float (to match the image):
```{unet.py}
    float_masks = tf.cast(masks, tf.float16)
    images_masks = tf.concat([images, float_masks], axis=-1)
```

As of today, I was unable to figure out the culprit for this bug. I already checked all values in the sweep's yaml file are valid parameters.

To reproduce, one may simply launch a run of `train.py` to get the mentioned error. Simply commenting the sweep commands and launching a "standalone" train results in a regular, fully-functional training.

The following fails:
```
    sweep_configs = yaml.safe_load((PKG_PATH / 'sweep_vnet.yaml').read_text())
    sweep_id = wandb.sweep(sweep=sweep_configs, project="galaxy-segmentation-project")
    wandb.agent(sweep_id, function=train)
    wandb.agent(f"ganegroup/galaxy-segmentation-project/{sweep_id}", function=train, count=29)
    # train()  # pylint: disable=no-value-for-parameter
```
The following works:
```
    # sweep_configs = yaml.safe_load((PKG_PATH / 'sweep_vnet.yaml').read_text())
    # sweep_id = wandb.sweep(sweep=sweep_configs, project="galaxy-segmentation-project")
    # wandb.agent(sweep_id, function=train)
    # wandb.agent(f"ganegroup/galaxy-segmentation-project/{sweep_id}", function=train, count=29)
    train()  # pylint: disable=no-value-for-parameter
```
